front-end-tools/src/components/WriteComponent.tsx

```diff
@@ -0,0 +1,618 @@
+                                            const file = files[0];
+                                            const arrayBuffer = await file.arrayBuffer();
+
+                                            const schema = btoa(
+                                                new Uint8Array(arrayBuffer).reduce((data, byte) => {
+                                                    return data + String.fromCharCode(byte);
+                                                }, '')
+                                            );
+                                            setUploadedModuleSchemaBase64(schema);
+                                        } else {
+                                            setUploadError('Upload schema file is undefined');
+                                        }
+                                    }}
+                                />
+                                <Form.Text />
+                            </Form.Group>
+                        )}
+
+                        {!deriveContractInfo && uploadedModuleSchemaBase64 && (
+                            <div className="actionResultBox">
+                                Schema in base64:
+                                <div>{uploadedModuleSchemaBase64.toString().slice(0, 30)} ...</div>
+                            </div>
+                        )}
+                        {uploadError !== undefined && <Alert variant="danger"> Error: {uploadError}. </Alert>}
+                        {error && <Alert variant="danger"> Error: {error}. </Alert>}
+                        {schemaError !== undefined && <Alert variant="danger"> Error: {schemaError}. </Alert>}
+                        {entryPointTemplate && (
+                            <>
+                                <br />
+                                <br />
+                                <div className="actionResultBox">
+                                    Parameter Template:
+                                    <pre>{JSON.stringify(JSON.parse(entryPointTemplate), undefined, 2)}</pre>
+                                </div>
+                            </>
+                        )}
+                        <br />
+                        <Form.Group className="mb-3">
+                            <Form.Label>Select input parameter type:</Form.Label>
+                            <Select
+                                options={INPUT_PARAMETER_TYPES_OPTIONS}
+                                {...form.register('inputParameterType')}
+                                onChange={(e) => {
+                                    form.setValue('inputParameterType', e?.value);
+                                    form.setValue('inputParameter', undefined);
+
+                                    setParsingError(undefined);
+                                }}
+                            />
+                            <Form.Text />
+                        </Form.Group>
```

Comment:
I don't understand why I have to choose something here when it's clear what the parameter must be. For example

![image](https://github.com/Concordium/concordium-smart-contract-tools/assets/105212/c55da780-a228-4d8b-b37b-b053baab89fe)


There the parameter is a string, that's what it says in the template. But I need to select the input parameter type as well. Why?

Reply:
How do you else want to do this programmatically? We have several templates but I am not aware of a good/global mapping between the templates and types. Hardcoding a long mapping here at the front end that I just defined seems also weird.